### PR TITLE
Updated Penn Events Tab but w/out Club Events (yet)

### PR DESCRIPTION
**Pull request by innocentoak**

### CODE REVIEW

File: PennMobile/Events/Views/PennEventsView.swift
Comment (anli5005) [inline PennMobile/Events/Views/PennEventsView.swift:20]: Can we get rid of this commented code

File: PennMobile/Events/Views/PennEventsView.swift
Comment (anli5005) [inline PennMobile/Events/Views/PennEventsView.swift:70]: We can use the `#Preview` macro in future. Won't block on this but just good to know

File: PennMobile/Events/Views/PennEventsViewerView.swift
Comment (anli5005) [inline PennMobile/Events/Views/PennEventsViewerView.swift:17]: Could also use

```swift
@Environment(\.dismiss) var dismiss
```

**Review by anli5005 — COMMENTED** [review]
Aside from a few nits, nice work!

Will defer to @JHawk0224 on anything brought up in the review

File: PennMobile/Events/Models/PennEvent.swift
Comment (JHawk0224) [inline PennMobile/Events/Models/PennEvent.swift:19]: Similar to what Anthony mentioned below what are the `start` and `end` fields?

File: PennMobile/Events/Models/PennEventsAPIManager.swift
Comment (JHawk0224) [reply PennMobile/Events/Models/PennEventsAPIManager.swift:19]: yes I agree

File: PennMobile/Events/Old Events/EventsAPI.swift
Comment (JHawk0224) [reply PennMobile/Events/Old Events/EventsAPI.swift:1]: Yes i think as long as these aren't used anywhere they should be deleted

File: PennMobile/Events/View Models/PennEventViewModel.swift
Comment (JHawk0224) [reply PennMobile/Events/View Models/PennEventViewModel.swift:11]: Yes I agree with Anthony, this should just be a struct. You can edit the `PennEvent` struct to use these default fields if you need and set the date formatter for decoding, and then just use that struct instead of this

File: PennMobile/Events/View Models/PennEventsViewModel.swift
Comment (JHawk0224) [inline PennMobile/Events/View Models/PennEventsViewModel.swift:46]: Small nit but maybe have an enum for this? This is fine though how it is

File: PennMobile/Events/Views/PennEventCellView.swift
Comment (JHawk0224) [inline PennMobile/Events/Views/PennEventCellView.swift:37]: Probably stay away from using `UIScreen.main.bounds.width`. Use a `GeometryReader` instead and get the width that way (so it's relative to the parent view)

File: PennMobile/Events/Views/PennEventCellView.swift
Comment (JHawk0224) [inline PennMobile/Events/Views/PennEventCellView.swift:25]: Maybe use `KFImage` instead, that way images should be cached and don't load each time

File: PennMobile/Events/Views/PennEventCellView.swift
Comment (JHawk0224) [inline PennMobile/Events/Views/PennEventCellView.swift:87]: As Anthony mentioned elsewhere you can also use `#Preview {`

File: PennMobile/Events/Views/PennEventsViewerView.swift
Comment (JHawk0224) [inline PennMobile/Events/Views/PennEventsViewerView.swift:32]: Probably use `PennCoordinate.shared.getDefaultRegion(at: )` instead for the default value for consistency. Look into this file and see if there's anywhere else where these helper methods might be useful.

File: PennMobile/Events/Views/PennEventsViewerView.swift
Comment (JHawk0224) [inline PennMobile/Events/Views/PennEventsViewerView.swift:39]: Maybe `KFImage` here too

File: PennMobile/Events/Views/PennEventsViewerView.swift
Comment (JHawk0224) [inline PennMobile/Events/Views/PennEventsViewerView.swift:46]: style nit lol

File: PennMobile.xcodeproj/project.pbxproj
Comment (anli5005) [reply PennMobile.xcodeproj/project.pbxproj:3305]: @innocentoak you didn't happen to check the "Automatically manage signing" checkbox right?

File: PennMobile/Events/Views/PennEventsViewerView.swift
Comment (JHawk0224) [reply PennMobile/Events/Views/PennEventsViewerView.swift:33]: Move to same shared location file

**Review by JHawk0224 — CHANGES_REQUESTED** [review]
Things todo:
- Remove commented file
- Combine Penn Coordinate with your location file
- Look into making the eventType an enum if possible
- Other unresolved comments

**Comment by JHawk0224** [comment]
I think this is good to merge in

**Review by JHawk0224 — APPROVED** [review]
I think this is good to merge in now

**Pull request merged**
